contracts/pool/pools/TokenLockingPoolBase.sol

```diff
@@ -0,0 +1,104 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.18;
+
+import "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "@openzeppelin/contracts-upgradeable/proxy/utils/Initializable.sol";
+import "@openzeppelin/contracts-upgradeable/access/OwnableUpgradeable.sol";
+import "@openzeppelin/contracts-upgradeable/security/ReentrancyGuardUpgradeable.sol";
+
+error InsufficientAmount();
+
+/**
+ * @title LockingPoolBase
+ * @dev Abstract contract providing basic functionality for token locking pools
+ */
+abstract contract TokenLockingPoolBase is
+    Initializable,
+    OwnableUpgradeable,
+    ReentrancyGuardUpgradeable
+{
+    using SafeERC20 for IERC20;
+
+    IERC20 internal endemicToken;
+    address feeReceiver;
```

Comment:
`feeReceiver` je trenutno nepromjenjiv pa cisto vidite je li to ok. I ajmo eksplicitno dat vidljivost ovoj varijabli.